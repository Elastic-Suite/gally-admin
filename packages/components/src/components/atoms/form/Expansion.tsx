import React from 'react'

import { IExpansions } from '@elastic-suite/gally-admin-shared'

import { FormHelperText, IconButton, InputLabel } from '@mui/material'
import InfoTooltip from './InfoTooltip'
import IonIcon from '../IonIcon/IonIcon'
import { StyledFormControl } from './InputText.styled'
import Button from '../buttons/Button'
import { InputTextError } from '../../index'
import { styled } from '@mui/system'
import { useTranslation } from 'next-i18next'
import TextFieldTagsError from './TextFieldTagsError'

const CustomPropError = ['error']
const CustomRoot = styled('div', {
  shouldForwardProp: (prop: string) => !CustomPropError.includes(prop),
})<{ error: boolean }>(({ theme, error }) => ({
  color: theme.palette.colors.neutral[900],
  fontFamily: 'var(--gally-font)',
  fontSize: theme.spacing(1.5),
  fontWeight: '600',
  lineHeight: '18px',
  padding: theme.spacing(2),
  border: '1px solid',
  borderRadius: theme.spacing(1),
  borderColor: error
    ? theme.palette.colors.primary[700]
    : theme.palette.colors.neutral[300],
  marginTop: theme.spacing(0.5),
  width: 'fit-content',
  '& .MuiFormLabel-root': {
    fontWeight: 500,
    color: theme.palette.colors.neutral[600],
  },
}))

export interface IProps {
  value: IExpansions
  onChange?: (value: IExpansions) => void
  label?: string
  required?: boolean
  fullWidth?: boolean
  margin?: 'none' | 'dense' | 'normal'
  infoTooltip?: string
  placeholder?: string
  error?: boolean
  helperText?: string
  helperIcon?: string
  showError?: boolean
}

interface IExpansionsFormatted {
  id: number | string
  referenceTerm: string
  terms: string[]
}

function iExpansionsFormattedToIExpansions(
  expansions: IExpansionsFormatted[]
): IExpansions {
  return expansions.map((expansion) => {
    const expansionItem =
      typeof expansion.id === 'string' ? { '@id': expansion.id } : {}
    return {
      ...expansionItem,
      referenceTerm: expansion.referenceTerm,
      terms: expansion.terms.map((term) => ({ term })),
    }
  })
}

function Expansion(props: IProps): JSX.Element {
  const {
    value,
    onChange,
    label,
    required,
    fullWidth,
    margin,
    infoTooltip,
    placeholder,
    error,
    helperText,
    helperIcon,
    showError,
  } = props

  if (value.length === 0) {
    onChange([{ referenceTerm: '', terms: [] }])
  }

  const { t } = useTranslation('thesaurus')
  let lastExpansionId = 0
  const expansions: IExpansionsFormatted[] =
    value && value instanceof Array && value.length > 0
      ? value.map((expansion) => ({
          id: expansion['@id'] ?? lastExpansionId++,
          referenceTerm: expansion.referenceTerm,
          terms: expansion.terms.map((term) => term.term),
        }))
      : []

  function addExpansion(): void {
    const expansionsUpdated = [...expansions]
    if (
      expansionsUpdated.length === 0 ||
      !(
        expansionsUpdated[expansionsUpdated.length - 1].referenceTerm === '' &&
        expansionsUpdated[expansionsUpdated.length - 1].terms.toString() ===
          [].toString()
      )
    ) {
      expansionsUpdated.push({
        id: lastExpansionId++,
        referenceTerm: '',
        terms: [],
      })
    }

    return onChange(iExpansionsFormattedToIExpansions(expansionsUpdated))
  }

  function updateTerms(
    expansionId: string | number,
    referenceTerm: string,
    terms: string[]
  ): void {
    const expansionsUpdated = [...expansions]
    const index = expansionsUpdated.findIndex(
      (expansionsUpdated) => expansionsUpdated.id === expansionId
    )
    if (index >= 0) {
      expansionsUpdated[index].terms = terms
      expansionsUpdated[index].referenceTerm = referenceTerm
    }
    return onChange(iExpansionsFormattedToIExpansions(expansionsUpdated))
  }

  function removeTerms(expansionId: string | number): void {
    const expansionsUpdated = [...expansions]
    const index = expansionsUpdated.findIndex(
      (expansion) => expansion.id === expansionId
    )

    if (index >= 0) {
      expansionsUpdated.splice(index, 1)
    }

    return onChange(iExpansionsFormattedToIExpansions(expansionsUpdated))
  }

  return (
    <StyledFormControl
      fullWidth={fullWidth}
      margin={margin}
      variant="standard"
      error={error}
    >
      {label || infoTooltip ? (
        <InputLabel shrink sx={{ maxWidth: '90%' }} required={required}>
          {label ? label : null}
          {infoTooltip ? <InfoTooltip title={infoTooltip} /> : null}
        </InputLabel>
      ) : undefined}
      <CustomRoot error={false}>
        <div
          style={{
            display: 'flex',
            gap: '8px',
            flexDirection: 'column',
            marginBottom: '24px',
          }}
        >
          {expansions.map((expansion) => (
            <div
              style={{ display: 'flex', alignItems: 'center' }}
              key={expansion.id}
            >
              <div
                style={{ position: 'relative', display: 'flex', gap: '32px' }}
              >
                <InputTextError
                  required
                  label={t('reference term')}
                  placeholder={t('reference term')}
                  value={expansion.referenceTerm}
                  onChange={(value): void => {
                    updateTerms(expansion.id, value as string, expansion.terms)
                  }}
                  showError={showError}
                />
                <TextFieldTagsError
                  required
                  label={t('expansion terms')}
                  fullWidth={fullWidth}
                  onChange={(terms): void => {
                    updateTerms(expansion.id, expansion.referenceTerm, terms)
                  }}
                  value={expansion.terms}
                  placeholder={placeholder}
                  additionalValidator={(value: string[]): string => {
                    if (value.length !== [...new Set(value)].length)
                      return 'synonymTermsDuplicate'
                    return ''
                  }}
                  // onRemoveItem={(): void => emptyTerms(expansion.id)}
                  showError={showError}
                />
              </div>
              <IconButton
                onClick={(): void => removeTerms(expansion.id)}
                style={{ margin: '24px 0 0 8px' }}
              >
                <IonIcon
                  style={{ fontSize: '18px', color: '#424880' }}
                  name="trash-outline"
                />
              </IconButton>
            </div>
          ))}
        </div>
        <div
          style={{ display: 'flex', justifyContent: 'right', marginRight: 44 }}
        >
          <Button
            size="large"
            display="secondary"
            endIcon={<IonIcon name="add" style={{ fontSize: 24 }} />}
            onClick={(): void => addExpansion()}
          >
            {t('add')}
          </Button>
        </div>
      </CustomRoot>
      {Boolean(helperText) && (
        <FormHelperText>
          {Boolean(helperIcon) && (
            <IonIcon
              name={helperIcon}
              style={{ fontSize: 18, marginRight: 2 }}
            />
          )}
          {helperText}
        </FormHelperText>
      )}
    </StyledFormControl>
  )
}

export default Expansion
