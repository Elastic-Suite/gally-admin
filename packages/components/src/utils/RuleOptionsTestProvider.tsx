import React, { ReactNode } from 'react'
import {
  IRuleEngineOperators,
  RuleAttributeType,
  getIri,
} from '@elastic-suite/gally-admin-shared'

import ruleEngineOperator from '../../public/mocks/rule_engine_operators.json'
import sourceFields from '../../public/mocks/source_fields.json'
import sourceFieldLabels from '../../public/mocks/source_field_labels.json'

import RuleOptionsProvider from '../components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider'

interface IProps {
  children: ReactNode
}

const sourceFieldLabelsMap = new Map(
  sourceFieldLabels['hydra:member'].map((label) => [
    label.sourceField,
    label.label,
  ])
)

const fields = sourceFields['hydra:member'].map((field) => {
  const label = sourceFieldLabelsMap.get(getIri('source_fields', field.id))
  return {
    id: field.id,
    code: field.code,
    label: (label || field.defaultLabel) ?? '',
    type: field.type as RuleAttributeType,
  }
})

function RuleOptionsTestProvider(props: IProps): JSX.Element {
  const { children } = props
  return (
    <RuleOptionsProvider
      catalogId={-1}
      localizedCatalogId={-1}
      fields={fields}
      ruleOperators={ruleEngineOperator as IRuleEngineOperators}
    >
      {children}
    </RuleOptionsProvider>
  )
}

export default RuleOptionsTestProvider
