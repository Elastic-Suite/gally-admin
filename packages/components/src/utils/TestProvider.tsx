import React, { ReactNode } from 'react'

import catalogs from '../../public/mocks/catalog.json'
import metadata from '../../public/mocks/metadata.json'

import { catalogContext, metadataContext } from '../contexts'
import { useCatalogs } from '../hooks'

import OptionsProvider from '../components/stateful-providers/OptionsProvider/OptionsProvider'

interface IProps {
  children: ReactNode
}

function TestProvider(props: IProps): JSX.Element {
  const { children } = props
  const contextValueCatalog = useCatalogs(catalogs['hydra:member'])
  const contextValueMetadata = metadata['hydra:member']

  return (
    <OptionsProvider>
      <catalogContext.Provider value={contextValueCatalog}>
        <metadataContext.Provider value={contextValueMetadata}>
          {children}
        </metadataContext.Provider>
      </catalogContext.Provider>
    </OptionsProvider>
  )
}

export default TestProvider
