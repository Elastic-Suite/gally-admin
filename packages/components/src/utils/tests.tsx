import React, { ReactNode } from 'react'
import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { Bundle, api } from '@elastic-suite/gally-admin-shared'

import { AppStore, RootState, setupStore } from '../store'

import AppProvider from '../components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: {
      'base_url/media': 'https://localhost/media/catalog/product/',
    },
    metadata: [
      {
        '@id': '/metadata/3',
        '@type': 'Metadata',
        id: 3,
        entity: 'product',
        sourceFields: ['/source_fields/217', '/source_fields/253'],
      },
      {
        '@id': '/metadata/4',
        '@type': 'Metadata',
        id: 4,
        entity: 'category',
        sourceFields: ['/source_fields/254', '/source_fields/286'],
      },
    ],
  },
  user: {
    requestedPath: '',
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
