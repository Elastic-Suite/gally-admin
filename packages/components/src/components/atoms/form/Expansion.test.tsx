import React from 'react'

import { renderWithProviders } from '../../../utils/tests'

import Expansion from './Expansion'
import { IExpansions } from '@elastic-suite/gally-admin-shared'
import { screen } from '@testing-library/react'
// import {fireEvent} from "@storybook/testing-library";

describe('Expansion match snapshot', () => {
  it('Expansion Full', () => {
    const expansions: IExpansions = [
      {
        '@id': '/thesaurus_expansions/1',
        '@type': 'ThesaurusExpansion',
        referenceTerm: 'top',
        terms: [
          {
            '@id': '/thesaurus_expansion_terms/1',
            '@type': 'ThesaurusSExpansionTerm',
            term: 'pants',
          },
          {
            '@id': '/thesaurus_expansion_terms/2',
            '@type': 'ThesaurusSExpansionTerm',
            term: 'trousers',
          },
        ],
      },
      {
        '@id': '/thesaurus_expansions/3',
        '@type': 'ThesaurusExpansion',
        referenceTerm: 'bottom',
        terms: [
          {
            '@id': '/thesaurus_expansion_terms/1',
            '@type': 'ThesaurusSExpansionTerm',
            term: 'ankle socks',
          },
          {
            '@id': '/thesaurus_expansion_terms/2',
            '@type': 'ThesaurusSExpansionTerm',
            term: 'socks',
          },
        ],
      },
    ]
    const { container } = renderWithProviders(
      <Expansion
        value={expansions}
        required
        fullWidth={false}
        margin="normal"
        infoTooltip="Expansion value ..."
        placeholder="Add text"
        error
        helperText="helperText"
        helperIcon="close"
      />
    )
    expect(container).toMatchSnapshot()
  })
})

describe('Expansion functional behavior', () => {
  it('Reference term empty error', () => {
    renderWithProviders(
      <Expansion value={[{ referenceTerm: '', terms: [{ term: 'pants' }] }]} />
    )
    expect(
      screen.getByText('expansion.error_message.reference_term.empty')
        .textContent
    ).toBe('expansion.error_message.reference_term.empty')
  })
  it('Expansion terms empty error', () => {
    renderWithProviders(
      <Expansion value={[{ referenceTerm: 'top', terms: [] }]} />
    )
    expect(
      screen.getByText('expansion.error_message.expansion_terms.empty')
        .textContent
    ).toBe('expansion.error_message.expansion_terms.empty')
  })
  it('Expansion terms term empty error', () => {
    renderWithProviders(
      <Expansion value={[{ referenceTerm: 'top', terms: [] }]} />
    )
    expect(
      screen.getByText('expansion.error_message.expansion_terms.empty')
        .textContent
    ).toBe('expansion.error_message.expansion_terms.empty')
  })
  it('Expansion terms duplicate error', () => {
    renderWithProviders(
      <Expansion
        value={[
          {
            referenceTerm: 'top',
            terms: [{ term: 'pants' }, { term: 'pants' }],
          },
        ]}
      />
    )
    expect(
      screen.getByText('expansion.error_message.expansion_terms.duplicate')
        .textContent
    ).toBe('expansion.error_message.expansion_terms.duplicate')
  })
})
