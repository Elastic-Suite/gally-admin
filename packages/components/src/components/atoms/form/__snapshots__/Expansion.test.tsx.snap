// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expansion match snapshot Expansion Full 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal css-1r71j8d-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-error Mui-required css-1cpkhc9-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      <span
        aria-label="Expansion value ..."
        class=""
        data-mui-internal-clone-element="true"
        style="display: inline-block; padding-left: 10px; margin-bottom: -5px;"
      >
        <ion-icon
          src="/images/ionicons/information-circle-outline.svg"
          style="color: rgb(129, 135, 185); width: 20px; height: 20px; display: flex;"
        />
      </span>
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error css-1ufxisk-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      class="css-cl0y5k"
    >
      <div
        style="display: flex; gap: 8px; flex-direction: column; margin-bottom: 24px;"
      >
        <div
          style="display: flex; align-items: center;"
        >
          <div
            style="position: relative; display: flex; gap: 32px;"
          >
            <div
              class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-rnc4nv-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
              >
                reference term
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ufxisk-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="InputText__Wrapper css-17fp4jk"
              >
                <div
                  class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1jf161a-MuiInputBase-root"
                  required=""
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
                    placeholder="reference term"
                    required=""
                    type="text"
                    value="top"
                  />
                </div>
              </div>
            </div>
            <div
              style="position: relative;"
            >
              <div
                class="css-vaijrj"
              >
                <div
                  style="margin-bottom: 4px;"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-error Mui-required css-rnc4nv-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                  >
                    expansion terms
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error css-1ufxisk-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                </div>
                <div
                  class="css-1mxyr7i"
                >
                  <div
                    class="css-r1vopi"
                  >
                    <div
                      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-69f6vo-MuiButtonBase-root-MuiChip-root"
                      role="button"
                      style="opacity: 1;"
                      tabindex="0"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        pants
                      </span>
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        data-testid="chipCloseButton"
                        tabindex="0"
                        type="button"
                      >
                        <ion-icon
                          src="/images/ionicons/close.svg"
                        />
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <div
                      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-69f6vo-MuiButtonBase-root-MuiChip-root"
                      role="button"
                      style="opacity: 1;"
                      tabindex="0"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        trousers
                      </span>
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        data-testid="chipCloseButton"
                        tabindex="0"
                        type="button"
                      >
                        <ion-icon
                          src="/images/ionicons/close.svg"
                        />
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <div
                      class="css-u4p24i"
                    >
                      <div
                        class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
                      >
                        <div
                          class="InputText__Wrapper css-17fp4jk"
                        >
                          <div
                            class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-z54rnl-MuiInputBase-root"
                          >
                            <input
                              aria-invalid="false"
                              class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
                              placeholder="Add text"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            style="margin: 24px 0px 0px 8px;"
            tabindex="0"
            type="button"
          >
            <ion-icon
              src="/images/ionicons/trash-outline.svg"
              style="font-size: 18px; color: rgb(66, 72, 128);"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          style="display: flex; align-items: center;"
        >
          <div
            style="position: relative; display: flex; gap: 32px;"
          >
            <div
              class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-rnc4nv-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
              >
                reference term
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ufxisk-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="InputText__Wrapper css-17fp4jk"
              >
                <div
                  class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1jf161a-MuiInputBase-root"
                  required=""
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
                    placeholder="reference term"
                    required=""
                    type="text"
                    value="bottom"
                  />
                </div>
              </div>
            </div>
            <div
              style="position: relative;"
            >
              <div
                class="css-vaijrj"
              >
                <div
                  style="margin-bottom: 4px;"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-error Mui-required css-rnc4nv-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                  >
                    expansion terms
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error css-1ufxisk-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                </div>
                <div
                  class="css-1mxyr7i"
                >
                  <div
                    class="css-r1vopi"
                  >
                    <div
                      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-69f6vo-MuiButtonBase-root-MuiChip-root"
                      role="button"
                      style="opacity: 1;"
                      tabindex="0"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        ankle socks
                      </span>
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        data-testid="chipCloseButton"
                        tabindex="0"
                        type="button"
                      >
                        <ion-icon
                          src="/images/ionicons/close.svg"
                        />
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <div
                      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-69f6vo-MuiButtonBase-root-MuiChip-root"
                      role="button"
                      style="opacity: 1;"
                      tabindex="0"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        socks
                      </span>
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        data-testid="chipCloseButton"
                        tabindex="0"
                        type="button"
                      >
                        <ion-icon
                          src="/images/ionicons/close.svg"
                        />
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <div
                      class="css-u4p24i"
                    >
                      <div
                        class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
                      >
                        <div
                          class="InputText__Wrapper css-17fp4jk"
                        >
                          <div
                            class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-z54rnl-MuiInputBase-root"
                          >
                            <input
                              aria-invalid="false"
                              class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
                              placeholder="Add text"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            style="margin: 24px 0px 0px 8px;"
            tabindex="0"
            type="button"
          >
            <ion-icon
              src="/images/ionicons/trash-outline.svg"
              style="font-size: 18px; color: rgb(66, 72, 128);"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        style="display: flex; justify-content: right; margin-right: 44px;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-156z0p6-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          add
          <span
            class="MuiButton-endIcon MuiButton-iconSizeLarge css-4egbtc-MuiButton-endIcon"
          >
            <ion-icon
              src="/images/ionicons/add.svg"
              style="font-size: 24px;"
            />
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium css-1kzozl2-MuiFormHelperText-root"
    >
      <ion-icon
        src="/images/ionicons/close.svg"
        style="font-size: 18px; margin-right: 2px;"
      />
      helperText
    </p>
  </div>
</div>
`;
